### PR TITLE
bismillah usecase

### PR DESCRIPTION
**Pull request by helloseyedjafari**

### CODE REVIEW

File: domain/src/commonMain/kotlin/com/islamversity/domain/repo/surah/GetSurahUsecase.kt [hunk 1]
```diff
@@ -1,26 +1,35 @@
...
 }
 
-class GetSurahsUsecaseImpl(
-    private val surahListRepo: SurahListRepo,
+class GetSurahUsecaseImpl(
+    private val surahRepo: SurahRepo,
     private val settingRepo: SettingRepo
-) : GetSurahsUsecase {
+) : GetSurahUsecase {
     override fun getSurahs() =
```
Comment (rezaiyan) [inline domain/src/commonMain/kotlin/com/islamversity/domain/repo/surah/GetSurahUsecase.kt:20]: plural `s` is needed

File: search/src/commonMain/kotlin/com/islamversity/search/SearchProcessor.kt [hunk 2]
```diff
@@ -46,7 +46,7 @@ class SearchProcessor(
                 SurahLocalModel(
                     surahID = it.id.id,
                     surahName = it.name,
-                    startingAyaOrder = 1
+                    startingAyaOrder = 1,
```
Comment (rezaiyan) [inline search/src/commonMain/kotlin/com/islamversity/search/SearchProcessor.kt:49]: It needs another `bismillahType` parameter

File: surah/src/commonMain/kotlin/com/islamversity/surah/SurahProcessor.kt [hunk 3]
```diff
@@ -23,17 +21,42 @@ class SurahProcessor(
...
+            .publish({
+                map {
+                    when (it.bismillahType) {
+                        BismillahType.NEEDED -> SurahResult.Bismillah.Show
+                        BismillahType.FIRST_AYA,
+                        BismillahType.NONE -> SurahResult.Bismillah.Hide
+                    }
+                }
+            }, {
+                TODO("process bismillah and get from the database if needed")
```
Comment (rezaiyan) [inline surah/src/commonMain/kotlin/com/islamversity/surah/SurahProcessor.kt:55]: seems backstage is ready

File: surah/src/main/res/layout/row_aya.xml [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+    android:layout_height="wrap_content"
+    android:orientation="horizontal"
+    android:weightSum="1">
+
+    <TextView
+        android:id="@+id/tvAyaNumber"
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:background="@android:color/darker_gray"
+        android:padding="8dp"
```
Comment (rezaiyan) [inline surah/src/main/res/layout/row_aya.xml:14]: resources

**Review by rezaiyan — APPROVED** [review]
Good job

File: search/src/commonMain/kotlin/com/islamversity/search/SearchProcessor.kt (hunk 2 above)
Comment (helloseyedjafari) [reply search/src/commonMain/kotlin/com/islamversity/search/SearchProcessor.kt:49]: wellcome to Kotlin 1.4!

File: surah/src/commonMain/kotlin/com/islamversity/surah/SurahProcessor.kt (hunk 3 above)
Comment (helloseyedjafari) [reply surah/src/commonMain/kotlin/com/islamversity/surah/SurahProcessor.kt:55]: will be next PR
to avoid making this lengthy

**Pull request merged**